### PR TITLE
Added sphinx/readthedocs documentation

### PR DESCRIPTION
**Pull request by isaac-tfn**
Describe Changes
---

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [ ] Updated RELEASE.md to reflect changes in PR
- [ ] Have new dependencies been added?
  - [ ] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [ ] Have they been added to `pyproject.toml`

### CODE REVIEW

**Comment by isaac-tfn** [comment]
Docs look good to me. Probably best to keep this PR open for now to make any changes that need making before MW

File: docs/Makefile
Comment (BenTaylor-TfN) [inline docs/Makefile:1]: Why do we need a makefile here, but not in https://github.com/Transport-for-the-North/caf.distribute/pull/11?

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+sys.path.insert(0, os.path.abspath(str(source)))
+sys.path.insert(
+    0,
+    os.path.abspath(r"C:\Users\IsaacScott\anaconda3\envs\toolkit\Lib\site-packages"),
+)
+
+project = "caf.toolkit"
+copyright = "2023, Transport for the North"
+author = "Benjamin Taylor"
```
Comment (BenTaylor-TfN) [inline docs/conf.py:35]:
```suggestion
author = "Transport for the North"
```

**Review by BenTaylor-TfN — COMMENTED** [review]
Really rough review. Avoided repeating myself until we've finished discussing some comments in https://github.com/Transport-for-the-North/caf.distribute/pull/11

File: docs/Makefile
Comment (isaac-tfn) [reply docs/Makefile:1]: I'm guessing you added this in your first try at docs and I didn't notice it was there/could be deleted!

Comment (BenTaylor-TfN) [reply docs/Makefile:1]: That sounds about right. Please can you delete, along with another makefile I saw - will tag it now with a comment to be clear

File: docs/make.bat
Comment (BenTaylor-TfN) [inline docs/make.bat:1]: Please remove

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+   contain the root `toctree` directive.
+
+Welcome to caf.toolkit's documentation!
+=======================================
+
+The Common Analytical Framework (CAF) Toolkit is a toolkit of transport
+planning and appraisal functionalities. It's the beginning of a project to make
+a lot of the useful stuff from
+`NorMITs Demand <https://github.com/Transport-for-the-North/NorMITs-Demand>`_
+more widely available and easily accessible.
```
Comment (BenTaylor-TfN) [inline docs/index.rst:13]:
```suggestion
The Common Analytical Framework (CAF) is a collection of transport planning and appraisal functionalities. It’s part of a project to make useful transport related functionality more widely available and easily accessible.

Tool Info
---------
CAF.Toolkit focusses on generic tools and functions that are utilised across the CAF Framework. Over time, alongside adding extras functionality to enhance other CAF packages, CAF.Toolkit will be populated with much of the generic functionality found in `NorMITs Demand <https://github.com/Transport-for-the-North/NorMITs-Demand>`_.
```

**Review by BenTaylor-TfN — APPROVED** [review]
